File: client-js/src/main/js/dcc/cellctl/Relation.js
/*
=================================================================
personium.io
Copyright 2014 FUJITSU LIMITED

Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and limitations under the License.
=================================================================
 */
/*global dcc:false */

///**
//* @class Relationのアクセスクラス.
//* @constructor
//* @augments dcc.AbstractODataContext
//*/
/**
 * It creates a new object dcc.cellctl.Relation.
 * @class This class represents Relation object.
 * @constructor
 * @augments dcc.AbstractODataContext
 * @param {dcc.Accessor} as Accessor
 * @param {Object} json
 */
dcc.cellctl.Relation = function(as, json) {
  this.initializeProperties(this, as, json);
};
dcc.DcClass.inherit(dcc.cellctl.Relation, dcc.AbstractODataContext);

///**
//* プロパティを初期化する.
//* @param {dcc.cellctl.Relation} self
//* @param {dcc.Accessor} as アクセス主体
//* @param {Object} json JSONオブジェクト
//*/
/**
 * This method initializes the properties of this class.
 * @param {dcc.cellctl.Relation} self
 * @param {dcc.Accessor} as Accessor
 * @param {Object} json JSON object
 */
dcc.cellctl.Relation.prototype.initializeProperties = function(self, as, json) {
  this.uber = dcc.AbstractODataContext.prototype;
  this.uber.initializeProperties(self, as);

///** クラス名. */
  /** Class name in camel case. */
  self.CLASSNAME = "Relation";

///** Relation名. */
  /** Relation Name. */
  self.name = null;
  /** _box.name. */
  self.boxname = null;

///** Roleとのリンクマネージャ. */
  /** Link manager with Role. */
  self.role = null;
///** ExtCellとのリンクマネージャ. */
  /** Link manager with ExtCell. */
  self.extCell = null;

  if (json !== null) {
    self.rawData = json;
    self.name = json.Name;
    self.boxname = json["_Box.Name"];
  }
  if (as !== undefined) {
    self.role = new dcc.cellctl.LinkManager(as, this, "Role");
    self.extCell = new dcc.cellctl.LinkManager(as, this, "ExtCell");
  }
};

///**
//* Relation名の設定.
//* @param {String} value Relation名
//*/
/**
 * This method sets the Relation name.
 * @param {String} value Relation name
 */
dcc.cellctl.Relation.prototype.setName = function(value) {
  this.name = value;
};

///**
//* Relation名の取得.
//* @return {String} Relation名
//*/
/**
 * This method gets the Relation name.
 * @return {String} Relation name
 */
dcc.cellctl.Relation.prototype.getName = function() {
  return this.name;
};

///**
//* _box.name値の設定.
//* @param {String} value _box.name値
//*/
/**
 * This method sets the box name.
 * @param {String} value _box.name value
 */
dcc.cellctl.Relation.prototype.setBoxName = function(value) {
  this.boxname = value;
};

///**
//* _box.name値の取得.
//* @return {String} _box.name値
//*/
/**
 * This method gets the box name.
 * @return {String} _box.name value
 */
dcc.cellctl.Relation.prototype.getBoxName = function() {
  return this.boxname;
};

///**
//* Relationオブジェクトのキーを取得する.
//* @return {String} ODataのキー情報
//*/
/**
 * This method gets the key for Relation.
 * @return {String} Key information
 */
dcc.cellctl.Relation.prototype.getKey = function() {
  if (this.boxname !== null) {
    return "(_Box.Name='" + this.boxname + "',Name='" + this.name + "')";
  } else {
    return "(_Box.Name=null,Name='" + this.name + "')";
  }
};

///**
//* クラス名をキャメル型で取得する.
//* @return {?} ODataのキー情報
//*/
/**
 * This method returns the class name.
 * @return {String} Class name
 */
dcc.cellctl.Relation.prototype.getClassName = function() {
  return this.CLASSNAME;
};

File: client-java/src/main/java/com/fujitsu/dc/client/Relation.java
/**
 * personium.io
 * Copyright 2014 FUJITSU LIMITED
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.fujitsu.dc.client;

import org.json.simple.JSONObject;

///**
// * Relationのアクセスクラス.
// */
/**
 * It creates a new object of Relation. This class represents Relation object.
 */
public class Relation extends AbstractODataContext {

    // /** クラス名. */
    /** Class name in camel case. */
    private static final String CLASSNAME = "Relation";

    // /** Relation名. */
    /** Relation Name. */
    private String name;
    /** _box.name. */
    private String boxname;

    // /** Roleとのリンクマネージャ. */
    /** Link Manager for Role. */
    public ODataLinkManager role;
    // /** ExtCellとのリンクマネージャ. */
    /** Link Manager for ExtCell. */
    public ODataLinkManager extCell;

    // /**
    // * コンストラクタ.
    // */
    /**
     * This is the default constructor.
     */
    public Relation() {
    }

    // /**
    // * コンストラクタ.
    // * @param as アクセス主体
    // */
    /**
     * This is the parameterized constructor with one argument and calling initialize method internally.
     * @param as Accessor
     */
    public Relation(final Accessor as) {
        this.initialize(as, null);
    }

    // /**
    // * コンストラクタ.
    // * @param as アクセス主体
    // * @param body 生成するAccountのJson
    // */
    /**
     * This is the parameterized constructor with two arguments and calling initialize method internally.
     * @param as Accessor
     * @param body Request Body
     */
    public Relation(final Accessor as, JSONObject body) {
        this.initialize(as, body);
    }

    // /**
    // * オブジェクトを初期化.
    // * @param as アクセス主体
    // * @param json サーバーから返却されたJSONオブジェクト
    // */
    /**
     * This method is use to initialize various class variables and parent class variables.
     * @param as Accessor
     * @param json Relation JSONObject
     */
    public void initialize(Accessor as, JSONObject json) {
        super.initialize(as);
        if (json != null) {
            rawData = json;
            name = (String) json.get("Name");
            boxname = (String) json.get("_Box.Name");
        }
        this.role = new ODataLinkManager(as, this);
        this.extCell = new ODataLinkManager(as, this);
    }

    // /**
    // * Relation名の設定.
    // * @param value Relation名
    // */
    /**
     * This method sets the RelationName value.
     * @param value RelationName
     */
    public void setName(String value) {
        this.name = value;
    }

    // /**
    // * Relation名の取得.
    // * @return Relation名
    // */
    /**
     * This method gets the RelationName value.
     * @return RelationName value
     */
    public String getName() {
        return name;
    }

    // /**
    // * _box.name値の設定.
    // * @param value _box.name値
    // */
    /**
     * This method sets the _box.name value.
     * @param value _box.name
     */
    public void setBoxName(String value) {
        this.boxname = value;
    }

    // /**
    // * _box.name値の取得.
    // * @return _box.name値
    // */
    /**
     * This method gets the _box.name value.
     * @return _box.name value
     */
    public String getBoxName() {
        return boxname;
    }

    // /**
    // * Relationオブジェクトのキーを取得する.
    // * @return ODataのキー情報
    // */
    /**
     * This method formats and returns the key value for Relation operations.
     * @return OData Key URL for Relation
     */
    public String getKey() {
        if (this.boxname != null) {
            return String.format("(_Box.Name='%s',Name='%s')", this.boxname, this.name);
        } else {
            return String.format("(_Box.Name=null,Name='%s')", this.name);
        }
    }

    // /**
    // * クラス名をキャメル型で取得する.
    // * @return ODataのキー情報
    // */
    /**
     * This method returns the Relation class name in camel case.
     * @return ClassName
     */
    public String getClassName() {
        return Relation.CLASSNAME;
    }

}
